scipy/stats/tests/test_multivariate.py

```diff
@@ -1964,6 +1965,40 @@ def test_to_corr(self):
         assert_allclose(m[0,0], 1)
 
 
+class TestRandomDirection:
+    @pytest.mark.parametrize("dim", [1, 2, 3, 4])
+    @pytest.mark.parametrize("size", [5, (5, 4)])
+    def test_samples(self, dim, size):
+        # test that samples have correct shape and norm 1
+        random_direction_dist = random_direction(dim)
+        samples = random_direction_dist.rvs(size)
+        if isinstance(size, int):
+            size = (size, )
+        expected_shape = size + (dim, )
+        assert samples.shape == expected_shape
+        norms = np.linalg.norm(samples, axis=-1)
+        expected_norms = np.ones(size)
+        assert_allclose(norms, expected_norms)
```

Comment:
```suggestion
        assert_allclose(norms, 1)
```

Another test would be to check the `mean_resultant` once that merges. The magnitude should be nearly zero.

You could also test for uniformity of the angle when projected onto a 2d plane. Had the random direction been generated using a uniform distribution instead of normal, you'd see non-uniformity.

```python3
import numpy as np
rng = np.random.default_rng(1638083107694713882823079058616272161)
import matplotlib.pyplot as plt

n = 100000

x = rng.uniform(-0.5, 0.5, (3, n))
th = np.arctan2(*x[1:])  # choose any two (or project onto any plane) 
plt.hist(th, alpha=0.5)

x = rng.standard_normal((3, n))
th = np.arctan2(*x[1:])
plt.hist(th, alpha=0.5)

plt.legend(('uniform', 'normal'))

```
![image](https://user-images.githubusercontent.com/6570539/197418896-e3a75ad8-a26b-4ac9-8c2c-6d1ed02568b7.png)